src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -93,7 +93,7 @@ public function validate(mixed $entity, Constraint $constraint)
                 throw new ConstraintDefinitionException(sprintf('The field "%s" is not mapped by Doctrine, so it cannot be validated for uniqueness.', $fieldName));
             }
 
-            if (property_exists(OrmClassMetadata::class, 'propertyAccessors')) {
+            if ($class instanceof OrmClassMetadata && property_exists($class, 'propertyAccessors')) {
```

Comment:
MongoDB ODM will most likely implement the same thing in the near future. We can make it future-proof by not checking the class of the metadata:

```suggestion
            if (property_exists($class, 'propertyAccessors')) {
```

The check is more contextual, which is more resilient.

Reply:
@GromNaN I suggest doctrine/persistence should add an API covering that need to allow writing code against the doctrine/persistence abstraction (right now, those `$class->propertyAccessors` or `$class->reflFields` are implementation-specific, which worked until now because all Doctrine projects were implementing the same approach, but it is not enforced by the doctrine/persistence abstraction).

Reply:
Thanks for your review @stof. I created https://github.com/doctrine/persistence/issues/426 to track this.

Also, I've remembered that both ORM and ODM has a `ClassMetadata::getFieldValue` method that should be used here (when it exists). An interface need to be added in doctrine/persistence to help detecting when this feature is implemented.